org.civicrm.reqangular/src/common/services/format-currency.service.js

```diff
@@ -2,85 +2,123 @@
-        // Add thousand seperator to amount using regex expression
-        var withThousandSeparator = amount.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1' + separators.thousand);
-        var formattedCurrency = withThousandSeparator.replace('.', separators.decimal);
-
-        return (+amount) ? formattedCurrency : ('0' + separators.decimal + '00');
+    * Adds defined thousand and decimal separators in the given amount
+    *
+    * @param {string} amount
+    * @return {promise}
+    */
+    function format (amount) {
+      return Settings.fetchSeparators().then(function (separators) {
+        return {
+          formatted: addSeparators(amount, separators),
+          unformatted: removeSeparators(amount, separators)
+        };
       });
     }
 
     /**
-     * Fetched the thousand and decimal seperators form backend
+     * Removes characters other than decimal separator
+     * Converts the formatted amount to numeric value (making calulable) by
+     * removing the thousand separators and replacing the decimal separator by " ."
      *
-     * @return {promise}
+     * @param  {string|Integer} amount
+     * @param  {object} separators
+     * @return {integer}
      */
-    function fetchSeparators () {
-      promise = promise || getSettings();
+    function removeSeparators (amount, separators) {
+      var expression = configureExpression(separators.decimal);
+      var strippedAmount = amount.toString().replace(new RegExp(expression, 'g'), '');
+      var amountWithDecimal = strippedAmount.replace(separators.decimal, '.');
 
-      return promise.then(function (result) {
-        return {
-          decimal: result.monetaryDecimalPoint,
-          thousand: result.monetaryThousandSeparator
-        };
-      });
+      return amountWithDecimal;
     }
 
     /**
-     * Gets defined settings data form backend
+     * Configure expression for special characters like \ and ]
      *
-     * @param  {Object} params
-     *
-     * @return {Promise}
+     * @param  {string} decimalSeparator
+     * @return {string}
      */
-    function getSettings () {
-      return api.sendGET('Setting', 'get').then(function (data) {
-        return data.values[0];
-      });
+    function configureExpression (decimalSeparator) {
+      var expression;
+
+      // Handle reserved regex characters that might be used as separators
+      switch (decimalSeparator) {
+        case '\\':
+          expression = '[^0-9\\\\]';
+          break;
+        case ']':
+          expression = '[^0-9\\]]';
+          break;
+        default:
+          expression = '[^0-9' + decimalSeparator + ']';
+      }
+
+      return expression;
     }
 
     /**
-     * Removes characters other than decimal separator
-     * Converts the formatted amount to numeric value (making calulable) by
-     * removing the thousand separators and replacing the decimal separator by " ."
+     * Formats the given amount with thousand separators
```

Comment:
the name of the func is `removeCharacters`, so I guess it does not format. It removes characters! :) Please amend the description to make it a bit more specific.

Reply:
Sorry! `* Formats the given amount with thousand separators` comment is for `addSeparators()` function not for `removeCharacters()` function.

Reply:
@bmnepali please make sure the descriptions are correct then

Reply:
ok sure!